Compilation error with `-Werror -Wdocumentation` due to broken `@param` documentation comments

TIL about `-Wdocumentation`. Looks like that checks the javadoc/doxygen style comments and complains about issues.
Yes.. build-failures in comments.

There are instances where the parameter names and the comments don't match up. In a project I'm building, this causes errors because of `-Werror`.

```
In file included from nxdk/lib/windows.h:8:
In file included from nxdk/lib/winapi/windows.h:11:
In file included from nxdk/lib/winapi/memoryapi.h:5:
nxdk/lib/xboxkrnl/xboxkrnl.h:2131:11: error: parameter 'Fill' not found in the function declaration [-Werror,-Wdocumentation]
 * @param Fill The ULONG-value with which the memory block will be filled
          ^~~~
nxdk/lib/xboxkrnl/xboxkrnl.h:2131:11: note: did you mean 'Pattern'?
 * @param Fill The ULONG-value with which the memory block will be filled
          ^~~~
          Pattern
nxdk/lib/xboxkrnl/xboxkrnl.h:2303:11: error: parameter 'String1' not found in the function declaration [-Werror,-Wdocumentation]
 * @param String1 Pointer to a null-terminated single-byte string
          ^~~~~~~
nxdk/lib/xboxkrnl/xboxkrnl.h:2303:11: note: did you mean 'String'?
 * @param String1 Pointer to a null-terminated single-byte string
          ^~~~~~~
          String
nxdk/lib/xboxkrnl/xboxkrnl.h:2451:16: error: empty paragraph passed to '@param' command [-Werror,-Wdocumentation]
 * @param update
   ~~~~~~~~~~~~^
nxdk/lib/xboxkrnl/xboxkrnl.h:2610:11: error: parameter 'The' not found in the function declaration [-Werror,-Wdocumentation]
 * @param The handle of the thread object to suspend.
          ^~~
nxdk/lib/xboxkrnl/xboxkrnl.h:2610:11: note: did you mean 'ThreadHandle'?
 * @param The handle of the thread object to suspend.
          ^~~
          ThreadHandle
nxdk/lib/xboxkrnl/xboxkrnl.h:2685:11: error: parameter 'The' not found in the function declaration [-Werror,-Wdocumentation]
 * @param The handle of the thread object to resume.
          ^~~
nxdk/lib/xboxkrnl/xboxkrnl.h:2685:11: note: did you mean 'ThreadHandle'?
 * @param The handle of the thread object to resume.
          ^~~
          ThreadHandle
```

--- a/lib/xboxkrnl/xboxkrnl.h
+++ b/lib/xboxkrnl/xboxkrnl.h
@@ -2128,7 +2128,7 @@ XBAPI VOID NTAPI RtlFreeAnsiString
  * Fills a specified memory area with repetitions of a ULONG value
  * @param Destination A pointer to the (ULONG-aligned) memory block which is to be filled
  * @param Length The length of the memory block which is to be filled
- * @param Fill The ULONG-value with which the memory block will be filled
+ * @param Pattern The ULONG-value with which the memory block will be filled
  */
 XBAPI VOID NTAPI RtlFillMemoryUlong
 (
@@ -2300,7 +2300,7 @@ XBAPI SIZE_T NTAPI RtlCompareMemory
 
 /**
  * Converts a single-byte character string (C-style string, NOT an ANSI_STRING object!) to an integer value
- * @param String1 Pointer to a null-terminated single-byte string
+ * @param String Pointer to a null-terminated single-byte string
  * @param Base Specifies the base (decimal, binary, octal, hexadecimal). If not given, the routine looks for prefixes in the given string (0x, 0o, 0b), default is decimal.
  * @param Value Pointer to a ULONG variable where the converted value will be stored
  * @return STATUS_SUCCESS if the string was successfully converted, STATUS_INVALID_PARAMETER otherwise
@@ -2447,8 +2447,8 @@ XBAPI NTSTATUS NTAPI PhyInitialize
 );
 
 /**
- * Read the status information from the NICs' registers
- * @param update
+ * Returns link status information either from NIC registers or from the last value cached by the kernel
+ * @param update If FALSE, the kernel returns the cached value, otherwise the hardware is polled and the cached value updated
  * @return Flags describing the status of the NIC
  */
 XBAPI DWORD NTAPI PhyGetLinkState
@@ -2607,7 +2607,7 @@ XBAPI VOID NTAPI NtUserIoApcDispatcher
 
 /**
  * Suspends the target thread and optionally returns the previous suspend count.
- * @param The handle of the thread object to suspend.
+ * @param ThreadHandle The handle of the thread object to suspend.
  * @param PreviousSuspendCount Optional pointer to a variable that receives the thread's previous suspend count.
  * @return The status of the operation.
  */
@@ -2682,7 +2682,7 @@ XBAPI NTSTATUS NTAPI NtSetEvent
 
 /**
  * Resumes the target thread (see NtSuspendThread) and optionally returns the previous suspend count.
- * @param The handle of the thread object to resume.
+ * @param ThreadHandle The handle of the thread object to resume.
  * @param PreviousSuspendCount Optional pointer to a variable that receives the thread's previous suspend count.
  * @return The status of the operation.
  */

